BlazarService/src/main/java/com/hubspot/blazar/guice/BlazarSlackModule.java

```diff
@@ -0,0 +1,77 @@
+package com.hubspot.blazar.guice;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import com.hubspot.blazar.base.visitor.RepositoryBuildVisitor;
+import com.hubspot.blazar.config.BlazarConfiguration;
+import com.hubspot.blazar.config.BlazarSlackConfiguration;
+import com.hubspot.blazar.listener.SlackDmNotificationVisitor;
+import com.hubspot.blazar.listener.SlackRoomNotificationVisitor;
+import com.hubspot.blazar.resources.SlackResource;
+import com.hubspot.blazar.resources.UserFeedbackResource;
+import com.hubspot.blazar.util.SlackClientWrapper;
+import com.hubspot.blazar.util.SlackMessageBuildingUtils;
+import com.ullink.slack.simpleslackapi.SlackSession;
+import com.ullink.slack.simpleslackapi.impl.SlackSessionFactory;
+
+/**
+ * This module handles the configuration of all slack-related features inside of Blazar.
+ * This module does not bind any slack related resources / clients / visitors etc. if slack is not configured.
+ */
+public class BlazarSlackModule implements Module {
+
+  public static final String SLACK_FEEDBACK_ROOM = "SLACK_FEDBACK_ROOM";
+  private static final Logger LOG = LoggerFactory.getLogger(BlazarSlackModule.class);
+  private final Optional<BlazarSlackConfiguration> slackConfiguration;
+  private final Multibinder<RepositoryBuildVisitor> repositoryBuildVisitorMultibinder;
+
+  public BlazarSlackModule(BlazarConfiguration configuration, Multibinder<RepositoryBuildVisitor> repositoryBuildVisitorMultibinder) {
+    this.slackConfiguration = configuration.getSlackConfiguration();
+    this.repositoryBuildVisitorMultibinder = repositoryBuildVisitorMultibinder;
+  }
+
+  @Override
+  public void configure(Binder binder) {
+
+    if (!slackConfiguration.isPresent()) {
+      LOG.info("Slack is not configured, not binding slack related resources or slack build notification visitors");
+      return;
+    }
+
+    binder.bind(SlackSession.class).toInstance(makeNewSlackSession());
```

Comment:
Because we're binding `toInstance` this will not create a new instance each time `@SlackSession` is injected. Because this is not `Managed` I didn't think that it would interact at all with the app lifecycle management that dropwizard does. 

This is from the documentation on `com.google.inject.Binder`:
```
... your module itself, not Guice, takes responsibility for obtaining an instance, then asks Guice to always use this single instance to fulfill all injection requests.
```

I can factor this out to a provides method annotated as Singleton.